docs/owo/ui/components/scroll-container.md

```diff
@@ -0,0 +1,63 @@
+---
+title: Scroll Container
+project: owo
+---
+
+# Scroll Container
+
+:material-xml: `#!xml <scroll>`<br>
+:material-language-java: `#!java Containers.verticalScroll(...)` | `#!java Containers.horizontalScroll(...)`
+
+The scroll container component creates a scrollable area for its child components. It supports vertical and horizontal scrolling, customizable scrollbars, and various sizing and alignment options.
+
+**Parameters:**
+
+- `direction` (required): The direction of the scroll container (`vertical` or `horizontal`).
+- `scrollbar-thiccness`: The thickness of the scrollbar in pixels.
+- `fixed-scrollbar-length`: The fixed length of the scrollbar in pixels.
+- `scrollbar-color`: The color of the scrollbar.
+- `scrollbar`: The scrollbar renderer to use for custom scrollbar styles.
+- `padding`: The padding around the child components.
+- `surface`: The background surface of the scroll container.
+- `horizontal-alignment`: The horizontal alignment of the child components.
+- `vertical-alignment`: The vertical alignment of the child components.
+- `allow-overflow`: Whether to allow child components to overflow the bounds of the scroll container.
+
+**Example (Code-driven):**
+
+```java
+Containers.verticalScroll(Sizing.fill(100), Sizing.content())
+    .child(Components.label(Text.literal("Item 1")))
+    .child(Components.label(Text.literal("Item 2")))
+    .child(Components.label(Text.literal("Item 3")))
+    .scrollbarThiccness(10)
+    .scrollbarColor(0xff0000)
+    .padding(Insets.of(10))
+    .surface(Surface.PANEL)
```

Comment:
Several changes on Scroll Containers:
- They now require at least one child component when constructed
- You can now specify what kind of Scrollbar you want using the `scrollbar()` method
- The color of the Scrollbar uses the `Color` helper

Another very important note is that a scroll container can not be content-sized vertically. This does not make sense, as you would not need to scroll if you can fit your entire content inside.

Example (doesn't use colored scrollbar):
```java
            Containers.verticalScroll(Sizing.content(), Sizing.fill(15), Containers.verticalFlow(Sizing.content(), Sizing.content())
                .child(Components.label(Text.literal("Item 1")))
                .child(Components.label(Text.literal("Item 2")))
                .child(Components.label(Text.literal("Item 2.1")))
                .child(Components.label(Text.literal("Item 2.2")))
                .child(Components.label(Text.literal("Item 2.3")))
                .child(Components.label(Text.literal("Item 2.4")))
                .child(Components.label(Text.literal("Item 3"))))
                .scrollbarThiccness(4)
                .scrollbar(ScrollContainer.Scrollbar.vanilla())
                .padding(Insets.of(10))
                .surface(Surface.PANEL)
```